### DISCUSSION TITLE
[Feature] SegmentedControl

### DISCUSSION THREAD

**Discussion by niels9001** [CommunityToolkit/Labs-Windows#314] [Proposals]
# Segmented Control

## Problem Statement

The WinUI controlset lacks a control that allows for configuring a view or setting. It allows developers to declare 2 to 5 items that can be (multi) selected to update a view or select a specific function.

Mobile platforms like iOS and Android have been using Segmented Control for years, but Windows does not. The Pivot control comes close, but hasn't been updated and is more used for navigation, not for selection.

## Inspiration

Windows 11 Settings -> Apps -> Installed apps
![image](https://user-images.githubusercontent.com/9866362/201341061-0c6a4f70-ab5a-407e-8a03-6ee0aefe93ff.png)

Snip and Sketch vNext
![image](https://user-images.githubusercontent.com/9866362/201341186-8f76796b-b6d9-4886-a9d0-0172afcedc36.png)

iOS
![image](https://user-images.githubusercontent.com/9866362/201478037-c6bb25df-7941-4b18-8d41-61eff744373a.png)

## Proposed Solution

- Simple and lightweight.
- Icon can be set
- Content can be set
- Multi-selection is supported
- By default, the first item is selected.
- Adhering to the Windows Visual Design specs to it feels native to WinUI.

```xml
<SegmentedControl>
      <SegmentedControlItem Icon="&#xEA3F;" Content="Item 1"/>
      <SegmentedControlItem Icon="&#xEA3F;" Content="Item 1"/>
      <SegmentedControlItem Icon="&#xEA3F;" Content="Item 1"/>
</SegmentedControl>
```

result:
![image](https://user-images.githubusercontent.com/9866362/201478364-d851e2b7-00fe-4ceb-9114-cc819b227641.png)

To support other use cases seen across Windows, we could support the following styles out of the box (better naming needed :)):

**DefaultSegmentedControlStyle**
![SegmentedControl1](https://user-images.githubusercontent.com/9866362/205310053-3bd34cae-e073-4b0a-a08f-11c523f20d16.gif)

**PillSegmentedControlStyle**
![SegmentedControl2](https://user-images.githubusercontent.com/9866362/205310101-3f671415-a291-4f14-9a52-724f69d4d858.gif)

**ToggleButtonSegmentedControlStyle**
![SegmentedControl3](https://user-images.githubusercontent.com/9866362/205310173-66492090-b409-46ed-b6a3-b9548f051555.gif)

## References
- https://www.figma.com/file/XjuZJySxR0ydkkYIrqk7ps/Windows-UI-3-(Community)?node-id=43251%3A144750&t=z6U37IDcuHdal6xA-4
- https://developer.apple.com/design/human-interface-guidelines/components/selection-and-input/segmented-controls
- https://github.com/microsoft/microsoft-ui-xaml/issues/2310 
- https://github.com/microsoft/microsoft-ui-xaml/issues/3759

**Comment by nikolayvpavlov** [comment]
I would humbly suggest the following more features:

1. Support for state icon: selected / not selected / highlighted.  

2. Support for rich content, i.e. not only text.  

```
<SegmentedControlItem Icon="&#xEA3F;">
    <SegmentedControlItem.Content>
         <StackPanel>
             ...
         </StackPane>
    </SegmentedControlItem.Content>
</SegmentedControlItem>
```

3. Orientation: vertical and horizontal.

4. Responsive: show icon only, or icon and content, wrappable.

**Comment by michael-hawker** [comment]
This was shipped in the [8.0 Release](https://github.com/CommunityToolkit/Windows/releases/tag/v8.0.230907), please update to use that version of the component. Any further issues/requests should start in the main repository.
